code/bngblaster/src/isis/isis_pdu.c

```diff
@@ -591,9 +591,13 @@ isis_pdu_add_tlv_ext_reachability(isis_pdu_s *pdu, uint8_t *system_id,
         /* set V and L flag always */
         *tlv_cur++ = 0x30;
         *tlv_cur++ = 0;
-        /* generate random adjacency SID but avoid reserved ranges 0-255 */
+        /* ensure proper random seed */
+        struct timespec ts;
+        clock_gettime(CLOCK_MONOTONIC, &ts);
+        srand((time_t)ts.tv_nsec);
```

Comment:
The pseudo random generator is already seed in main function. 
https://github.com/rtbrick/bngblaster/blob/002fcb7a3a5d1aa0e9b95a2cdfd265a7434387ae/code/bngblaster/src/bbl.c#L412

Reply:
I obviously overlooked this in bbl. 

I found srand(time(0)); to weak, only goes for seconds. In lspgen case I several times got duplicate randoms.

Reply:
Since you can't restart the BNG Blaster in less than a second, the default seed (`srand(time(0));`) should be fine. You should also not re-seed with every time we call `isis_pdu_add_tlv_ext_reachability`. Therefore I would propose to remove this here.

Reply:
Even when using the optimal seed value, there remains a potential risk of two adjacencies sharing the same SID. To mitigate this issue, I propose assigning a unique SID directly within the isis_adjacency_init() function. This would involve enhancing the isis_adjacency_s structure by introducing a new member specifically designed to store this unique SID. By doing so, each adjacency will have its own distinct SID, eliminating any possibility of conflicts and ensuring reliable identification of adjacencies within the system.

I propose initializing the starting SID using a pseudo-random generator and then incrementing it with each new adjacency. To avoid reserved ranges, the SID should be incremented as follows: `SID++`. If the SID falls below 256 or exceeds 4095, it should be reset to 256, and the process should continue from there. This approach ensures that each adjacency receives a unique SID while avoiding conflicts with reserved values.

Reply:
Agree, we need proper fix for this. Since its also part of isis_lsp_self_update() it will also generate new adj-sid every time we refresh LSP.

Reply:
I would suggest to change the approach and move to static init value for adjacency-sid defined by user (lsp read-config-file/bbl config) and we only SID++ this.

Reply:
I want to rephrase:

- on lspgen I would suggest to change the approach and move to static value assignment for adjacency-sid for each link always (user must ensure its not overlapping for same node)
- on bbl I would change the boolean knob adjacency-sid to a static init value, based on this value we count up (SID++) for each link 

This eliminate the use of rand() and ensure its always deterministic.

If you agree, I would move forward with this.

Reply:
Incremental is fine as it is deterministic. Please go ahead.